osidb/models/affect.py

```diff
@@ -172,6 +172,26 @@ class Meta:
     def __str__(self):
         return str(self.uuid)
 
+    def save(self, *args, **kwargs):
+        # Check that either purl or ps_component exists
+        if not self.purl and not self.ps_component:
+            raise ValidationError(
+                f"Affect ({self.uuid}) for {self.ps_module} must have either purl or ps_component."
+            )
```

Comment:
This is not the right place for the validation. A validation method should be defined for it